### DISCUSSION TITLE
where did Microsoft.Graph.Models.Json go to in V5.56?

### DISCUSSION THREAD

**Discussion by tomvandyckVdP** [microsoftgraph/msgraph-sdk-dotnet#2587] [Q&A]
hi, 
since I upgraded to V5.56 the type Json in namspace Microsoft.Graph.Models no longer exists.
Where did this go to? Has it been replaced?

I was using this type in combination with a custom Newtonsoft.Json.JsonConverter: in the (overridden) ReadJson method I did a reader.Skip in order to skip Json objects, since JsonSoft does not seem to be serializing Json typed objects ...

**Comment by andrueastman** [comment]
Thanks for raising this @tomvandyckVdP 

The type was removed as this was a bug in the SDK generator that was blocking the representation of properties that were not typed. As the `Microsoft.Graph.Models.Json` was an object type, users could not deserialize data in the models that were arrays or primitive types therefore leading to loss of data. 

These instances were replaced with instances of UntypedNode which are documented at https://learn.microsoft.com/en-us/openapi/kiota/serialization?tabs=csharp#untyped-node

**Comment by RogerTimmerman** [comment]
Hi, 
I have something related to the question of @tomvandyckVdP :
We recently upgraded our application from Microsoft Graph SDK v4 to v5 and found an annoying issue in v5 in serialization of email messages with file-attachments to Json string format after receipt of an email via the Graph API's.
The serialization is performed and stored and later deserialize the Json to a email message again and e.g. forward that email including extra information.
- Using JsonSerializer the contentbytes of the file is missing from the attachments property in the Json output string (looks like the issue Tom addresses)
- Using KiotaJsonSerializer all email properties are missing in Json output string except the attachments property (does contain the contentbytes)

The actual questions are:
- Can somebody provide me a (small) C# (within .Net framework 4.8) code example how to tackle this issue as the Microsoft documentation (about untyped nodes etc.) does not help me that much.
- Also I wonder whether a Json file serialized within SDK 4 can be deserialized directly within SDK 5.

Replies are appreciated,
With regards
Roger Timmerman
